### PR TITLE
cmake-no-system, add windows & aarch64 support

### PR DESCRIPTION
**Pull request by dholth**
Adapt build-cmake-with-binary from cmake-feedstock

### CODE REVIEW

**Comment by chenghlee** [comment]
> Could we build `zstd` in the same way that it was already built for aarch64 & s390x?
> 
> ```
> Loading channels: done
> # Name                       Version           Build  Channel             
> zstd                           1.4.8      h8d054b9_1  pkgs/main           
> zstd                           1.4.9      h8d054b9_1  pkgs/main   ```
> ```

zstd for s390x was [built using `cmake-no-system`](https://github.com/AnacondaRecipes/zstd-feedstock/blob/master/recipe/meta.yaml#L29-L31). I didn't build the aarch64 version, but I suspect it was either cmake-no-system or (the now obsolete) cmake-binary (repack).

File: recipe/meta.yaml [hunk 1]
```diff
@@ -34,17 +49,17 @@ about:
...
+    - cmake/Utilities/cmcurl/COPYING
+    - cmake/Utilities/cmexpat/COPYING
+    - cmake/Utilities/cmjsoncpp/LICENSE
+    - cmake/Utilities/cmlibarchive/COPYING
+    - cmake/Utilities/cmliblzma/COPYING
+    - cmake/Utilities/cmlibrhash/COPYING
+    - cmake/Utilities/cmlibuv/LICENSE
+    - cmake/Utilities/cmzlib/Copyright.txt
+    - cmake/Utilities/cmzstd/LICENSE
   summary: CMake built without system libraries for use when building CMake dependencies.
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:63]: - No documentation url provided
- No development url provided

Comment (dholth) [reply recipe/meta.yaml:65]: added doc_url, dev_url

**Review by skupr-anaconda — APPROVED** [review]
I think someone else should review it too

File: recipe/meta.yaml [hunk 2]
```diff
@@ -13,16 +13,22 @@ source:
...
+    folder: cmake-bin  # [win32]
+
   - url: https://cmake.org/files/v{{ version[:4] }}/cmake-{{ version }}-win64-x64.zip  # [win64]
     sha256: 70812d07a19d4e14a9e625fdd698f0dcd0c6f32d81d63979c645f5327e0772e2  # [win64]
-    # applied to win32/win64:
-    folder: cmake-bin  # [win]
+    folder: cmake-bin  # [win64]
 
 build:
   number: 1
```
Comment (katietz) [inline recipe/meta.yaml:27]: Please bump build-number

File: recipe/meta.yaml [hunk 3]
```diff
@@ -13,16 +13,22 @@ source:
...
+  skip: True  # [win and vc<14]
+  detect_binary_files_with_prefix: False
+  ignore_run_exports:
+    - vc
```
Comment (katietz) [inline recipe/meta.yaml:31]: is this ignore_run_exports required?

**Review by katietz — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml (hunk 3 above)
Comment (dholth) [reply recipe/meta.yaml:31]: I think there are many errors if it is not present. Copied from the cmake-feedstock.

**Pull request merged**